Cabal/src/Distribution/PackageDescription/Check.hs

```diff
@@ -1589,20 +1589,35 @@ checkPaths pkg =
   ++
   [ PackageDistInexcusable $
       GlobSyntaxError "data-files" (explainGlobSyntaxError pat err)
-  | pat <- dataFiles pkg
-  , Left err <- [parseFileGlob (specVersion pkg) pat]
+  | (Left err, pat) <- zip globsDataFiles $ dataFiles pkg
   ]
   ++
   [ PackageDistInexcusable
       (GlobSyntaxError "extra-source-files" (explainGlobSyntaxError pat err))
-  | pat <- extraSrcFiles pkg
-  , Left err <- [parseFileGlob (specVersion pkg) pat]
+  | (Left err, pat) <- zip globsExtraSrcFiles $ extraSrcFiles pkg
   ]
   ++
   [ PackageDistInexcusable $
       GlobSyntaxError "extra-doc-files" (explainGlobSyntaxError pat err)
-  | pat <- extraDocFiles pkg
-  , Left err <- [parseFileGlob (specVersion pkg) pat]
+  | (Left err, pat) <- zip globsExtraDocFiles $ extraDocFiles pkg
+  ]
+  ++
+  [ PackageBuildWarning $
```

Comment:
Is this really a sintax error? Or a dangerous glob?

Reply:
It's not a syntax error. It should probably go to different constructor

Reply:
So, this is amended now? @ffaf1, could you perhaps give it another look?

Reply:
@ffaf1: ping?

Reply:
Moved it to separate constructor